### ISSUE TITLE
Submit Form to Update Interest Groups of Existing Subscriber (Ignore Errors)

### ISSUE THREAD

**Issue by vusisindane** [yikesinc/yikes-inc-easy-mailchimp-extender#479]
I want people who already exist in my list to have an option to subscribe to an interest group.  

If a subscriber already exists when submitting the form, they get this error 

> To update your MailChimp profile, please click to send yourself an update link.

But this prevents them from being redirected to a link, which I've setup, after they signup because the form views this as an error.  

I just want the plugin to ignore the fact that user already exists and update them so they can access the re-direct link as well.

How can I set it up to achieve this?

P.S

If I'm not mistaken, there used to be an option to ignore errors and update the subscriber.  But now it provides a link.

**Comment by TheTracyL** [comment]
Hello.

We added emailing the update link to the user, because it is a security issue to allow folks to update profiles without permission. All a person would need is another user's email address to update all their profile info.

Thank you!
-Tracy

**Comment by EvanHerman** [comment]
Hi Vusisindane,

As Tracy has mentioned, this was a security feature that was requested by a number of users. A lot of people viewed this as a security issue, and we leaned towards agreeing with them so we implemented a two factor update system.

However, this can be easily adjusted by using one of the filters built into the plugin. You'll want to use the filter `yikes-mailchimp-user-subscribe-api-request` that we have included, which allows you to alter any of the API data in the request being sent to MailChimp. (https://github.com/yikesinc/yikes-inc-easy-mailchimp-extender/blob/master/plugin_core/public/partials/shortcodes/process/process_form_submission.php#L137-L150)

Here is a sample function that should allow your users to submit the forms and not receive the double optin mesage. Keep in mind, you'll want to add this snippet to your active themes **functions.php** file, or if using a child theme, your child themes **functions.php** file.


```php
/**
* Set the API request to update_existing => 1
*/
function custom_mailchimp_api_request( $api_data, $form_id, $list_id, $user_email ) {
   $api_data[ 'update_existing' ] = 1;
   return $api_data;
}
add_filter( 'yikes-mailchimp-user-subscribe-api-request', 'custom_mailchimp_api_request', 10, 4 );
```

Let me know if that works for you!


Evan
